internal/verify/run_test_suite.go

```diff
@@ -347,11 +351,17 @@ func runTest(ctx context.Context, eng Checker, test *policyv1.Test, action strin
 	return details
 }
 
-func performCheck(ctx context.Context, eng Checker, inputs []*enginev1.CheckInput, options *policyv1.TestOptions, trace bool) ([]*enginev1.CheckOutput, []*enginev1.Trace, error) {
+func performCheck(ctx context.Context, eng Checker, inputs []*enginev1.CheckInput, options *policyv1.TestOptions, trace bool) (_ []*enginev1.CheckOutput, traces []*enginev1.Trace, _ error) {
 	var checkOpts []engine.CheckOpt
 
+	usedDefaultNow := false
 	if now := options.GetNow(); now != nil {
 		checkOpts = append(checkOpts, engine.WithNowFunc(now.AsTime))
+	} else {
+		checkOpts = append(checkOpts, engine.WithNowFunc(func() time.Time {
+			usedDefaultNow = true
+			return time.Time{}
```

Comment:
AFAIU, this code snippet helps to fail the misconfigured test, but the test must first be error-free. Can the call to `Check` result in an error because time isn't deterministic? I think it can.

Reply:
I'm not sure how the return value of `now()` being the zero time could cause a `Check` call to error out. Even if it caused a CEL expression to error, at worst that error would cause a rule not to be activated, wouldn't it? It wouldn't be propagated back to the caller.

Is there a scenario you're thinking of where this wouldn't need the case?

Reply:
No, there shouldn't be a problem with zero value. My bad.